atcoder/maxflow.py

```diff
@@ -0,0 +1,209 @@
+from typing import NamedTuple, Optional, List
+
+
+class MaxFlow:
+    class Edge(NamedTuple):
+        src: int
+        dst: int
+        cap: int
+        flow: int
+
+    class _Edge:
+        dst: int
+        cap: int
+        rev: Optional['MaxFlow._Edge']
+
+        def __init__(self, dst: int, cap: int):
+            self.dst = dst
+            self.cap = cap
+            self.rev = None
+
+    def __init__(self, n: int):
+        self._n = n
+        self._g: List[List[MaxFlow._Edge]] = [[] for _ in range(n)]
+        self._edges: List[MaxFlow._Edge] = []
+
+    def add_edge(self, src: int, dst: int, cap: int) -> int:
+        assert 0 <= src < self._n
+        assert 0 <= dst < self._n
+        assert 0 <= cap
+        m = len(self._edges)
+        e = MaxFlow._Edge(dst, cap)
+        re = MaxFlow._Edge(src, 0)
+        e.rev = re
+        re.rev = e
+        self._g[src].append(e)
+        self._g[dst].append(re)
+        self._edges.append(e)
+        return m
+
+    def get_edge(self, i: int) -> Edge:
+        assert 0 <= i < len(self._edges)
+        e = self._edges[i]
+        re = e.rev
+        return MaxFlow.Edge(
+            re.dst,
+            e.dst,
+            e.cap + re.cap,
+            re.cap
+        )
+
+    def edges(self) -> List[Edge]:
+        return [self.get_edge(i) for i in range(len(self._edges))]
+
+    def change_edge(self, i: int, new_cap: int, new_flow: int):
+        assert 0 <= i < len(self._edges)
+        assert 0 <= new_flow <= new_cap
+        e = self._edges[i]
+        e.cap = new_cap - new_flow
+        e.rev.cap = new_flow
+
+    def flow(self, s: int, t: int, flow_limit: Optional[int] = None) -> int:
+        assert 0 <= s < self._n
+        assert 0 <= t < self._n
+        assert s != t
+        if flow_limit is None:
+            flow_limit = sum(e.cap for e in self._g[s])
```

Comment:
Note: Used the sum of capacity of the edges leaving `s` instead of `numeric_limits<T>::max()`.